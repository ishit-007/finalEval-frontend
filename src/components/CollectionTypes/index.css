.collection-types {
  height: 100vh;
  width: 30vw;
  background-color: #2a2929;
}

.collection-types .heading {
  display: flex;
  align-items: center;
  background-color: #4e0eae;
  color: white;
  height: 8%;
  padding-left: 20px;
}

.collection-types-header{
  padding-left: 20px;
  padding-right: 20px;
  display: flex;
  flex-wrap: wrap;
  color: white;
  justify-content: space-between;
}

.collection-types-list ul{
  font-size:x-large;
  color: #939393;
}

.collection-types-list ul li{
  padding-top: 15px;
}

.builder{
  font-size:x-large;
  color:white;
  margin-top: 35px;
  background-color: black;
  padding: 5px;
}

@media only screen and (max-width: 660px) {
  .collection-types-list ul {
    font-size: x-small;
  }
}

@media only screen and (max-width: 450px) {
  .collection-types-header h2{
    display: none;
  }
}