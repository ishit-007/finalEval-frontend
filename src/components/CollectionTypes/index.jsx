/* eslint-disable no-unused-vars */
import './index.css';
import magnifyingGlass from '../../assets/icon-search-dark.png';
import React from 'react';
import axios from 'axios';
import { useNavigate } from 'react-router-dom';
import PropTypes from 'prop-types';
import LoginPage from '../../pages/LoginPage';

const CollectionTypes = props => {
  const [contentTypes, setContentTypes] = React.useState([]);
  const [selectedContentData, setSelectedContentData] = React.useState([]);
  const navigate = useNavigate();

  React.useEffect(() => {
    const token = localStorage.getItem('token');
    axios
      .get('http://localhost:8080/contentTypes/all', {
        headers: {
          token: token,
        },
      })
      .then(resp => {
        if (resp.data.length > 0) {
          const filteredData = resp.data.filter(contentType => {
            if (contentType.name === 'CompanyProfile') {
              return true;
            } else {
              return false;
            }
          });
          setSelectedContentData(filteredData[0]);
        }
        setContentTypes(resp.data);
      })
      .catch(err => {
        console.log(err);
      });
  }, []);

  return (
    <div className="collection-types">
      <div className="heading">
        <h1>CMS+</h1>
      </div>
      <div className="collection-list">
        <div className="collection-types-header">
          <h2>COLLECTION TYPES</h2>
          <img src={magnifyingGlass} alt="" />
        </div>
        <div className="collection-types-list">
          <ul className="collection-types-list">
            {contentTypes.map(contentType => {
              return (
                <li
                  key={contentType.id}
                  onClick={() => {
                    navigate(`/content-types/${contentType.id}`);
                  }}>
                  {contentType.name}
                </li>
              );
            })}
          </ul>
        </div>
        <div className="builder">
          <p
            onClick={() => {
              navigate('/content-types');
            }}>
            CONTENT TYPE BUILDER
          </p>
        </div>
      </div>
    </div>
  );
};

export default CollectionTypes;
