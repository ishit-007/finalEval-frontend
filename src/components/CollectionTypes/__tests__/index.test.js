import { render } from '@testing-library/react';
import React from 'react';
import CollectionTypes from '..';
jest.mock('react-router-dom');

describe('CollectionTypes', () => {
  it('should render the CollectionTypes component', async () => {
    const { asFragment } = render(<CollectionTypes />);
    expect(asFragment()).toMatchSnapshot();
    // expect(queryByText('CMS+')).toBeTruthy();
  });
});