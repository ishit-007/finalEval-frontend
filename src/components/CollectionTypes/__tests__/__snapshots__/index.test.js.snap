// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionTypes should render the CollectionTypes component 1`] = `
<DocumentFragment>
  <div
    class="collection-types"
  >
    <div
      class="heading"
    >
      <h1>
        CMS+
      </h1>
    </div>
    <div
      class="collection-list"
    >
      <div
        class="collection-types-header"
      >
        <h2>
          COLLECTION TYPES
        </h2>
        <img
          alt=""
          src="icon-search-dark.png"
        />
      </div>
      <div
        class="collection-types-list"
      >
        <ul
          class="collection-types-list"
        />
      </div>
      <div
        class="builder"
      >
        <p>
          CONTENT TYPE BUILDER
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
